cpp/src/arrow/memory_pool.h

```diff
@@ -36,43 +37,58 @@ namespace internal {
 // Helper tracking memory statistics
 
 class MemoryPoolStats {
- public:
-  MemoryPoolStats() : bytes_allocated_(0), max_memory_(0) {}
-
-  int64_t max_memory() const { return max_memory_.load(); }
-
-  int64_t bytes_allocated() const { return bytes_allocated_.load(); }
-
-  int64_t total_bytes_allocated() const { return total_allocated_bytes_.load(); }
+ private:
+  // All atomics are updated according to Acquire-Release ordering.
+  // https://en.cppreference.com/w/cpp/atomic/memory_order#Release-Acquire_ordering
+  //
+  // max_memory_, total_allocated_bytes_, and num_allocs_ only go up (they are
+  // monotonically increasing) which can allow some optimizations.
+  std::atomic<int64_t> bytes_allocated_{0};
```

Comment:
Should it aligned by cache-line for performance ( https://en.cppreference.com/w/cpp/thread/hardware_destructive_interference_size )?

Reply:
I did put all the atomics on the same cache line, but I used `64` instead of this STL thing that will grow like the inflation.